pkg/machine/hyperv/machine.go

```diff
@@ -762,8 +762,18 @@ func (m *HyperVMachine) startHostNetworking() (string, machine.APIForwardingStat
 	c := cmd.Cmd(gvproxyBinary)
 
 	if logrus.IsLevelEnabled(logrus.DebugLevel) {
-		c.Stdout = os.Stdout
-		c.Stderr = os.Stderr
+		homedir, err := os.UserHomeDir()
+		if err != nil {
+			return "", 0, fmt.Errorf("obtain user home dir: %w", err)
+		}
+		gvproxyLog, err := os.Create(filepath.Join(homedir, "gvproxy.log"))
+		if err != nil {
+			return "", 0, fmt.Errorf("create log file: %w", err)
+		}
+		defer gvproxyLog.Close()
+
+		c.Stdout = gvproxyLog
+		c.Stderr = gvproxyLog
```

Comment:
Can I just say that I hate having applications write directly into the home dir. Shouldn't this be but into the machine dir itself? At the very least so it gets removed when you remove the machine but also so it doesn't conflict if we can ever start more than one machine at once.

Second because you duplicate the same code twice can you split this into a function and please add a log line that says logging output from X to file Y. I have no idea how much these programs will write but I can imagine someone running out of disk space if they are not aware of this.

Reply:
What gets logged here? For CRC we have a similar log file and this can be a lot. We get a NTP (UDP) timeout a few times once in a while. This causes our log to balloon to several megabytes for a day.